### PR TITLE
fix: remove warning for compile error

### PR DESCRIPTION
**Pull request by wep21**
Signed-off-by: wep21 <border_goldenmarket@yahoo.co.jp>

## Description(required)

Note: This compile error occurs only in my environment, so I create this PR as a draft.
There are 3 errors below.
`/usr/include/pcl-1.10/pcl/octree/octree_key.h:155:9: error: ISO C++ prohibits anonymous structs [-Werror=pedantic]`
`/opt/ros/galactic/include/rcutils/logging_macros.h:72:18: error: ‘filter_limit_negative’ may be used uninitialized in this function [-Werror=maybe-uninitialized]`
`/opt/ros/galactic/include/rcutils/logging_macros.h:72:18: error: ‘keep_organized’ may be used uninitialized in this function [-Werror=maybe-uninitialized]`

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Read [commit-guidelines][commit-guidelines]
- [ ] Assign PR to reviewer

If you are adding new package following items are required:

- [ ] Documentation with description of the package is available
- [ ] A sample launch file and parameter file are available if the package contains executable nodes

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [x] Commits are properly organized and messages are according to the guideline
- [x] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[commit-guidelines]: https://www.conventionalcommits.org/en/v1.0.0/
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: sensing/pointcloud_preprocessor/CMakeLists.txt [hunk 1]
```diff
@@ -8,7 +8,7 @@ if(NOT CMAKE_CXX_STANDARD)
   set(CMAKE_CXX_EXTENSIONS OFF)
 endif()
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  add_compile_options(-Wall -Wextra -Wpedantic -Werror)
+  add_compile_options(-Wall -Wextra -Werror)
```
Comment (kenji-miyake) [inline sensing/pointcloud_preprocessor/CMakeLists.txt:11]: Why did you remove `-Wpedantic`?

**Comment by kenji-miyake** [comment]
@wep21 Could you write the error log in the description? :pray:

File: sensing/pointcloud_preprocessor/CMakeLists.txt (hunk 1 above)
Comment (wep21) [reply sensing/pointcloud_preprocessor/CMakeLists.txt:11]: `/usr/include/pcl-1.10/pcl/octree/octree_key.h:155:9: error: ISO C++ prohibits anonymous structs [-Werror=pedantic]`
This causes the error above. I don't have any idea why `-Wno-gnu-anonymous-struct` doesn't work...

**Comment by wep21** [comment]
@kenji-miyake I added the errors in the description. Please check it out.

File: sensing/pointcloud_preprocessor/CMakeLists.txt (hunk 1 above)
Comment (kenji-miyake) [reply sensing/pointcloud_preprocessor/CMakeLists.txt:11]: I see. Hmm... :thinking:

File: sensing/pointcloud_preprocessor/src/passthrough_filter/passthrough_filter_uint16_nodelet.cpp [hunk 2]
```diff
@@ -94,25 +94,27 @@ rcl_interfaces::msg::SetParametersResult PassThroughFilterUInt16Component::param
...
-  bool keep_organized;
-  get_param(p, "keep_organized", keep_organized);
-  if (impl_.getKeepOrganized() != keep_organized) {
-    RCLCPP_DEBUG(
-      get_logger(), "Setting the filter keep_organized value to: %s.",
-      keep_organized ? "true" : "false");
-    // Call the virtual method in the child
-    impl_.setKeepOrganized(keep_organized);
+  bool keep_organized{};
+  if (get_param(p, "keep_organized", keep_organized)) {
```
Comment (kenji-miyake) [inline sensing/pointcloud_preprocessor/src/passthrough_filter/passthrough_filter_uint16_nodelet.cpp:98]: Why did you add this `if`? I guess if `get_param` fails `keep_organized` will be `false`.

Comment (wep21) [reply sensing/pointcloud_preprocessor/src/passthrough_filter/passthrough_filter_uint16_nodelet.cpp:98]: The failure of `get_param` means that parameter is not updated, so set parameter function of `impl_` should be blocked by this `if` condition.

Comment (wep21) [reply sensing/pointcloud_preprocessor/src/passthrough_filter/passthrough_filter_uint16_nodelet.cpp]: Simplify at https://github.com/autowarefoundation/autoware.universe/pull/198/commits/faaa97a3544affaf15cb6cf4f342c645293c43ef.

File: common/had_map_utils/CMakeLists.txt [hunk 3]
```diff
@@ -5,27 +5,31 @@ cmake_minimum_required(VERSION 3.5)
...
-  include/had_map_utils/had_map_query.hpp
-  include/had_map_utils/had_map_visualization.hpp
-  include/had_map_utils/visibility_control.hpp
   src/had_map_utils.cpp
   src/had_map_computation.cpp
   src/had_map_conversion.cpp
   src/had_map_query.cpp
   src/had_map_visualization.cpp)
 
+target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC
```
Comment (kenji-miyake) [inline common/had_map_utils/CMakeLists.txt:28]: Don't you need to keep this comment? `# Disable warnings due to external dependencies`

File: planning/behavior_velocity_planner/CMakeLists.txt [hunk 4]
```diff
@@ -56,6 +56,12 @@ ament_auto_add_library(scene_module_lib SHARED
   src/utilization/interpolate.cpp
 )
 
+target_include_directories(scene_module_lib
+  SYSTEM PUBLIC
```
Comment (kenji-miyake) [inline planning/behavior_velocity_planner/CMakeLists.txt:60]: Why do you add `PUBLIC` for only a few packages? What is the difference between the packages that you didn't add it?

File: common/had_map_utils/CMakeLists.txt (hunk 3 above)
Comment (wep21) [reply common/had_map_utils/CMakeLists.txt:29]: fixed at https://github.com/autowarefoundation/autoware.universe/pull/198/commits/7be9c892ce497f820b56917ed471747f81aa5d7b.

File: planning/behavior_velocity_planner/CMakeLists.txt (hunk 4 above)
Comment (wep21) [reply planning/behavior_velocity_planner/CMakeLists.txt:60]: `PUBLIC/PRIVATE/INTERFACE` keyword is necessary for `target_include_directories`.
https://cmake.org/cmake/help/latest/command/target_include_directories.html

Comment (kenji-miyake) [reply planning/behavior_velocity_planner/CMakeLists.txt:60]: Ah, I see, thank you!

**Review by kenji-miyake — APPROVED** [review]

**Comment by ajay1606** [comment]
@wep21 @kenji-miyake 

Hello, While building the **main** branch from the Autoware architecture proposal, I also encountered the following errors that were mentioned here a few days ago. Anyhow after a few workarounds able to overcome those errors; Please check this out if it's helpful for you also.
```
/usr/include/pcl-1.10/pcl/octree/octree_key.h:155:9: error: ISO C++ prohibits anonymous structs [-Werror=pedantic]
/opt/ros/galactic/include/rcutils/logging_macros.h:72:18: error: ‘filter_limit_negative’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
/opt/ros/galactic/include/rcutils/logging_macros.h:72:18: error: ‘keep_organized’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
```
1. Include "SYSTEM" in the include directory in CMakeLists as below; (/pointcloud_preprocessor/CMakeLists.txt)

```
include_directories(
  include
)
include_directories(SYSTEM
  ${Boost_INCLUDE_DIRS}
  ${PCL_INCLUDE_DIRS}
  ${EIGEN3_INCLUDE_DIRS}
  ${OpenCV_INCLUDE_DIRS}
  ${GRID_MAP_INCLUDE_DIR}
)
```

2. While building workspace add  (-DCMAKE_CXX_FLAGS="-w")
colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_FLAGS="-w"

Regards,
Ajay

**Comment by kenji-miyake** [comment]
@ajay1606 Thank you for sharing that. Yes, we also used `-w`. :+1:
